backend/api/Services/AutoScheduleService.cs

```diff
@@ -75,5 +102,341 @@ MissionDefinition missionDefinition
+                    Name = missionDefinition.Name,
+                    Robot = robot,
+                    MissionId = missionDefinition.Id,
+                    Status = MissionStatus.Pending,
+                    MissionRunType = MissionRunType.Normal,
+                    DesiredStartTime = DateTime.UtcNow,
+                    Tasks = missionTasks,
+                    InstallationCode = missionDefinition.InstallationCode,
+                    InspectionArea = missionDefinition.InspectionArea,
+                };
+
+                if (missionRun.Tasks.Any())
+                {
+                    missionRun.SetEstimatedTaskDuration();
+                }
+
+                await missionRunService.Create(missionRun);
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, "{ErrorMessage}", e.Message);
+                return;
+            }
+            return;
+        }
+
+        public async Task<AutoScheduleFrequency?> UpdateAutoScheduleFrequency(
+            MissionDefinition missionDefinition,
+            AutoScheduleFrequency? newAutoScheduleFrequency
+        )
+        {
+            if (missionDefinition.AutoScheduleFrequency is null && newAutoScheduleFrequency is null)
+            {
+                return null;
+            }
+            if (
+                missionDefinition.AutoScheduleFrequency is not null
+                && newAutoScheduleFrequency is not null
+            )
+            {
+                if (
+                    missionDefinition.AutoScheduleFrequency.TimesOfDayCET.SequenceEqual(
+                        newAutoScheduleFrequency.TimesOfDayCET
+                    )
+                    && missionDefinition.AutoScheduleFrequency.DaysOfWeek.SequenceEqual(
+                        newAutoScheduleFrequency.DaysOfWeek
+                    )
+                )
+                {
+                    return missionDefinition.AutoScheduleFrequency;
+                }
+            }
+
+            if (missionDefinition.AutoScheduleFrequency?.AutoScheduledJobs is not null)
+            {
+                await SkipAllAutoMissions(missionDefinition);
+            }
+
+            if (newAutoScheduleFrequency is null)
+            {
+                missionDefinition.AutoScheduleFrequency = null;
+                return (
+                    await missionDefinitionService.Update(missionDefinition)
+                ).AutoScheduleFrequency;
+            }
+            missionDefinition.AutoScheduleFrequency ??= new AutoScheduleFrequency();
+
+            missionDefinition.AutoScheduleFrequency.TimesOfDayCET =
+                newAutoScheduleFrequency.TimesOfDayCET;
+            missionDefinition.AutoScheduleFrequency.DaysOfWeek =
+                newAutoScheduleFrequency.DaysOfWeek;
+
+            await StartJobsForMissionDefinition(missionDefinition);
+            return (await missionDefinitionService.Update(missionDefinition)).AutoScheduleFrequency;
+        }
```

Comment:
```suggestion
            await StartJobsForMissionDefinition(missionDefinition);
            var updatedMissionDefinition = await missionDefinitionService.Update(missionDefinition)
            return updatedMissionDefinition.AutoScheduleFrequency;
        }
```

This is a preference but i personally think the code gets more readable